src/snakemake/scattergather.py

```
@@ -0,0 +1,138 @@
+                raise WorkflowError(
+                    "Unsupported scatter process options: "
+                    + ", ".join(sorted(map(str, unknown)))
+                )
+            if "items" not in value:
+                raise WorkflowError(
+                    "Scatter process mappings require an 'items' entry."
+                )
+            if "impatient" in value and "tolerance" not in value:
+                raise WorkflowError(
+                    "Scatter process 'impatient' mode requires a 'tolerance' bound."
+                )
```

Ruff (0.15.5)

[warning] Avoid specifying long messages outside the exception class

(TRY003)